src/resolve/resolved-dns-rr.c

```diff
@@ -182,6 +182,23 @@ bool dns_resource_key_is_dnssd_ptr(const DnsResourceKey *key) {
                 dns_name_endswith(dns_resource_key_name(key), "_udp.local");
 }
 
+bool dns_resource_key_is_dnssd_two_label_ptr(const DnsResourceKey *key) {
+        assert(key);
+
+        /* Check if this is a PTR resource key used in Service Instance
+         * Enumeration as described in RFC6763 § 4.1, excluding selective
+         * service names described in RFC6763 § 7.1. */
```

Comment:
Not that it matters much but I think it would be better to add a link to https://datatracker.ietf.org/doc/html/rfc6763#section-9 because that's where it's mentioned explicitly that the rdata of those PTR RRs should be two labels plus a domain name.